robot/impl/local_robot.go

```diff
@@ -1209,3 +1210,28 @@ func (r *localRobot) CloudMetadata(ctx context.Context) (cloud.Metadata, error)
 	md.MachinePartID = cloud.ID
 	return md, nil
 }
+
+// restartSingleModule constructs a single-module diff and calls updateResources with it.
+func (r *localRobot) restartSingleModule(ctx context.Context, mod config.Module) error {
+	diff := config.Diff{
+		Left:     r.Config(),
+		Right:    r.Config(),
+		Added:    &config.Config{},
+		Modified: &config.ModifiedConfigDiff{Modules: []config.Module{mod}},
+		Removed:  &config.Config{},
+	}
+	return r.manager.updateResources(ctx, &diff)
```

Comment:
I like this approach (instead of updating the module from underneath the rest of the robot).

Reply:
I initially tried calling the module's `Reconfigure` directly, but it removed the resources + then didn't add them back -- `updateResources` did the right thing, and it respects the existing mutexes

Reply:
I'm worried just calling `updateResources` won't work in some circumstances, but let's chat offline.

Reply:
I was wrong; I think `updateResources` is correct here and what you have is good.